doc/source/ray-air/examples/batch_tuning.ipynb

```diff
@@ -0,0 +1,1197 @@
+    "    df[\"dropoff_location_id\"] = df[\"dropoff_location_id\"].fillna(-1)\n",
+    "    return df"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "72369cbf-e346-441c-b7e3-9280ccabc7f6",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "## Define your Ray Tune Search Space and Search Algorithm <a class=\"anchor\" id=\"define_search_space\"></a>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e310b13e-ba83-4c47-84e6-d1e916483b2b",
+   "metadata": {},
+   "source": [
+    "**First, define a search space of experiment to run.** \n",
+    "> The search space together with a search algorithm determine which experiments will be run.  \n",
+    "\n",
+    "Common search algorithms include grid search, random search, Bayesian search, Hyperopt, and Optuna.  For more details, see [Working with Tune Search Spaces](https://docs.ray.io/en/master/tune/tutorials/tune-search-spaces.html#tune-search-space-tutorial).  Deciding the best combination of search space and search algorithm is part of the art of being a Data Scientist and depends on the data, algorithm, and problem being solved!  \n",
+    "\n",
+    "**Below, we define our search space is:**\n",
+    "- 2 different Scikit-learn algorithms \n",
+    "- Some or all NYC taxi drop-off locations. \n",
+    "\n",
+    "**And we define the search algorithm is:**\n",
+    "- Grid search.\n",
+    "\n",
+    "> This means every permutation of each algorithm and each NYC Taxi drop-off location will be a separate experiment!  \n",
+    "\n",
+    "Ray Tune partitions the Search space using the specified Search algorithm and takes care of running a Tune job on each partition in parallel.  Specifically, Ray Tune will pass a config dictionary to each partition and make a Trainable function call."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "id": "7de03814-4eec-4363-a623-790f040b99be",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# 1. Define a search space.\n",
+    "sample_locations = [145, 166, 152] if SMOKE_TEST else all_location_ids\n",
+    "search_space = {\n",
+    "    \"model\": tune.grid_search([LinearRegression(fit_intercept=True), \n",
+    "                                DecisionTreeRegressor(max_depth=3)]),\n",
+    "    \"location\": tune.grid_search(sample_locations),\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "28814e79-f53e-4527-a959-e952f7d57fe5",
+   "metadata": {},
+   "source": [
+    "## Define a Trainable (callable) function <a class=\"anchor\" id=\"define_trainable\"></a>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "cdb3c9a4-b2df-4063-8c04-f5e0136c90bf",
+   "metadata": {},
+   "source": [
+    "🧪 Typically when you are running Data Science experiments, you want to be able to keep track of summary metrics for each experiment, so you can decide at the end which experiments were the best.  That way, you can decide which model to deploy.\n",
```

Comment:
let's make sure to start off with what's relevant to the workload here (we define the trainable in order to train and evaluate a sklearn model on a data partition) and then move to details about metrics (because we want to report the error and checkpoint the model for reuse)